calrissian/k8s.py

```diff
@@ -46,6 +47,7 @@ def __init__(self):
     def submit_pod(self, pod_body):
         pod = self.core_api_instance.create_namespaced_pod(self.namespace, pod_body)
         log.info('Created k8s pod name {} with id {}'.format(pod.metadata.name, pod.metadata.uid))
+        PodMonitor.add(pod)
```

Comment:
Is there a possibility that a SIGTERM arrives just before this method, the main thread runs `delete_pods()` and the pod created here is leaked?

Reply:
Shouldn't we test that `PodMonitor.add` is called when users run `submit_pod`